File: Python/Wrapping_Paper.py
#! usr/bin/python

f=open('dimensions.txt', 'r')
ribbon=0
wrapping=0
for line in f:
	for l in line.split("\n "):
		l,w,h=map(int, l.split("x"))
		wrapping += 2*l*w + 2*w*h + 2*h*l + min(l*w, w*h, h*l)
		ribbon += (2*l*w*h - 2* max(l,w,h)) + l*w*h
print "Wrapping needed: " + str(wrapping) + " Ribbon needed: " +  str(ribbon)

File: Java/Wrapping_paper.java
package daytwo;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Wrapping_Paper {


	public static void main(String[] args) {
		new Wrapping_Paper().squareFeetOfWrappingPaper(new File("inputs/input2.txt"));;
	}

	public void squareFeetOfWrappingPaper (File file) {
		int totalArea = 0;
		int ribbon = 0; 
		try {
			Scanner scan = new Scanner(file);
			String line;
			while (scan.hasNextLine()) {
				line = scan.nextLine();
				String [] tokens = line.split("x");

				int	l = Integer.parseInt(tokens[0]);
				int	w = Integer.parseInt(tokens[1]);
				int	h = Integer.parseInt(tokens[2]);

	

				ribbon+=2*l+2*w+2*h - 2 * Math.max(h, Math.max(w, l));
				ribbon+=l*w*h;

				totalArea+=(2*l*w+2*w*h+2*h*l);
				totalArea+=Math.min(l*w, Math.min(w*h, h*l));

			}
			System.out.println("Total of wrapper: " + totalArea);
			System.out.println("Total of ribbon: " + ribbon);
			scan.close();

		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}

	}
}
